src/announcement/announcement.ts

```diff
@@ -0,0 +1,81 @@
+//eslint-disable-next-line
+require("dotenv").config();
+import Web3 from "web3";
+import { Contract } from "web3-eth-contract";
+import { AbiItem } from "web3-utils";
+
+import { KeccakHash } from "../types/hash";
+import { hashPrefix } from "../utilities/hashPrefix";
+
+const BATCH_CONTRACT_ADDRESS = process.env.BATCH_CONTRACT_ADDRESS as string;
+
+export interface BatchArguments {
+  provider: Web3;
+  accountAddress: string;
+  uri: string;
+  hash: KeccakHash;
+}
+const BATCH_CONTRACT_ABI: AbiItem[] = [
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: false,
+        internalType: "bytes32",
+        name: "hash",
+        type: "bytes32",
+      },
+      {
+        indexed: false,
+        internalType: "string",
+        name: "dsnpUri",
+        type: "string",
+      },
+    ],
+    name: "DSNPBatch",
+    type: "event",
+  },
+  {
+    inputs: [
+      {
+        internalType: "bytes32",
+        name: "hash",
+        type: "bytes32",
+      },
+      {
+        internalType: "string",
+        name: "dsnpUri",
+        type: "string",
+      },
+    ],
+    name: "batch",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+const contract: Contract | null = null;
+
+export const getContract = (web3Instance: Web3): Contract => {
+  if (contract) return contract;
+  return new web3Instance.eth.Contract(BATCH_CONTRACT_ABI, BATCH_CONTRACT_ADDRESS);
+};
+
+/**
+ * batch() allows users call the batch smart contract and post the URI and hash
+ * of a generated batch to the blockchain.
+ *
+ * @param provider  The web3 instance used for calling the smart contract
+ * @param accountAddress   The address from which to post the batch
+ * @param uri       The URI of the hosted batch to post
+ * @param hash      A hash of the batch contents for use in verification
+ * @returns         A [web3 contract receipt promise](https://web3js.readthedocs.io/en/v1.3.4/web3-eth-contract.html#id36)
+ */
+export const batch = async ({ provider, accountAddress, uri, hash }: BatchArguments): Promise<any> => {
+  const contract = (await getContract(provider)) as Contract;
+  return await contract.methods.batch(hashPrefix(hash), uri).send({
+    from: accountAddress,
+    gas: 27000,
```

Comment:
I know why we hardcoded this, but we really don't want to hard code this. Do we want to switch to using estimate gas instead? (I guess we don't really want to do that in cases where the node is processing the transaction however. Hmm....)

Reply:
`estimateGas` doesn't seem to work properly with POA nodes in the current version of web3.js. It returns a value exactly at the amount of gas needed which causes `send()` to throw `Signer Error: gas limit is too low. Need at least 25496`. We could use the estimate and add an extra `1000`, but then we'd be right back at having a constant in the method.

Reply:
I think I prefer estimate + a constant small overage. The reason being is that different contract implementations may have different gas costs.